### DISCUSSION TITLE
Your feedback is requested! Polls related to .NET6 TShock!

### DISCUSSION THREAD

**Discussion by hakusaro** [Pryaxis/TShock#2517] [General]
We've asked a couple questions on Twitter and on Discord. Please share your thoughts here, too.

1. We're considering making the primary executable for TShock `TShock.exe` instead of `TerrariaServerAPI.exe` or `TerrariaServer.exe`. This would simplify installation but potentially create confusion for people who are familiar with the system.
2. We're considering shipping in "standalone" mode, which means that you wouldn't need to install the .NET framework or Mono to run TShock. If we do this, however, the download package would be significantly larger and contain many DLL files for the framework to run.

For simplicity, please vote in the accompanying Twitter post or Discord polls if you can, or post here with your thoughts. It'll be simpler if we can tally two places, though, so, when replying to this post on GitHub, please do so in such a way that it's obvious what you're voting for and why you're voting for it. I trust you all to be clever and self-organize in a cohesive and logical way.
* https://twitter.com/pryaxis/status/1464850612445597697
* https://twitter.com/pryaxis/status/1464851344662020105

We aren't overly committed to any of these ideas. I'm posting because @DeathCradle presented these options to me, and I wanted to get everyone's thoughts before we make a decision.

**Comment by comdar16** [comment]
What about plugins? Will old plugins work on .NET6

**Reply by hakusaro** [reply]
Thanks to some magic from @DeathCradle, probably.

**Reply by SignatureBeef** [reply]
In known/simple cases, maybe with a migration tool*
Most of the tsapi plugin api remains unchanged however there is small changes to the itile namespace, and also depends on what dependencies plugins use and if they work on net6.
For the most part it shouldn't be hard to migrate plugins either, but we can explore the migration tool as time permits.

**Reply by comdar16** [reply]
> In known/simple cases, maybe with a migration tool* Most of the tsapi plugin api remains unchanged however there is small changes to the itile namespace, and also depends on what dependencies plugins use and if they work on net6. For the most part it shouldn't be hard to migrate plugins either, but we can explore the migration tool as time permits.

Sure.

If you will need help with plugin update you can ping me or something so i could try to help you. Becuase i'v been updating some old plugins for new Tshock versions in the past.

**Comment by CyberBotX** [comment]
I briefly mentioned this on Discord but figured I should elaborate on things here. (I already voted on my choice on Discord.)

Is this going to be an all-or-nothing thing as far as requiring .NET Framework and Mono is concerned? I use FreeBSD for my TShock servers, and while there is currently no official .NET 6 runtime/SDK for FreeBSD currently, there is an unofficial one at https://github.com/Servarr/dotnet-bsd that I have found works quite when for basic .NET 6 usage. I know that there is a long-standing issue on the .NET GitHub about FreeBSD support, but it seems as if support just isn't there yet.

Because the version I'm using is unofficial, I have no idea how it handles the situation of having Mono also installed alongside it. To top it off, the version of Mono in FreeBSD's ports tree is not the latest (they have 3 different versions of Mono 5.x [although 2 of them are identical] and the only 6.x version they have available is 6.8.0). I've never had to run a program that uses both .NET and .NET Framework and as far as I can tell, there does not appear to be a way to run a .NET Framework executable file via the `dotnet` command. (I tried and it doesn't work as expected because of the lack of the files needed to make it be seen as either "self-contained" or "framework-dependent".)

**Reply by hakusaro** [reply]
> Because the version I'm using is unofficial, I have no idea how it handles the situation of having Mono also installed alongside it. To top it off, the version of Mono in FreeBSD's ports tree is not the latest (they have 3 different versions of Mono 5.x [although 2 of them are identical] and the only 6.x version they have available is 6.8.0).

Just to clarify you're currently using Mono, right?

**Reply by CyberBotX** [reply]
Yes, currently I am running TShock via Mono.
